gateway/testutil_test.go

```diff
@@ -0,0 +1,21 @@
+package gateway
+
+import (
+	"runtime"
+	"testing"
+	"time"
+)
+
+func TestStartTestLeaks(t *testing.T) {
+	t.Logf("Started with %d goroutines", runtime.NumGoroutine())
+
+	ts := StartTest(nil)
+	ts.Close()
+
+	tick := func() {
+		time.Sleep(time.Second)
+		t.Logf("Tick with %d goroutines", runtime.NumGoroutine())
+	}
+
+	tick()
+}
```

Comment:
**Suggestion:** Consider using a more descriptive test name. The current name, 'TestStartTestLeaks', does not clearly indicate what the test is checking.
```suggestion
func TestGoroutineLeakOnStartTest(t *testing.T) {
	t.Logf("Started with %d goroutines", runtime.NumGoroutine())

	ts := StartTest(nil)
	ts.Close()

	tick := func() {
		time.Sleep(time.Second)
		t.Logf("Tick with %d goroutines", runtime.NumGoroutine())
	}

	tick()
}
```